src/renderer/kube-object.store.ts

```diff
@@ -267,31 +269,69 @@ export abstract class KubeObjectStore<T extends KubeObject = any> extends ItemSt
   }
 
   subscribe(apis = this.getSubscribeApis()) {
-    let disposers: {(): void}[] = [];
-
-    const callback = (data: IKubeWatchEvent) => {
-      if (!this.isLoaded) return;
-
-      this.eventsBuffer.push(data);
-    };
+    const abortController = new AbortController();
+    const namespaces = [...this.loadedNamespaces];
 
-    if (this.context.cluster?.isGlobalWatchEnabled && this.loadedNamespaces.length === 0) {
-      disposers = apis.map(api => api.watch({
-        namespace: "",
-        callback: (data) => callback(data),
-      }));
+    if (this.context.cluster?.isGlobalWatchEnabled && namespaces.length === 0) {
+      apis.forEach(api => this.watchNamespace(api, "", abortController));
     } else {
-      apis.map(api => {
+      apis.forEach(api => {
         this.loadedNamespaces.forEach((namespace) => {
-          disposers.push(api.watch({
-            namespace,
-            callback: (data) => callback(data)
-          }));
+          this.watchNamespace(api, namespace, abortController);
         });
       });
     }
 
-    return () => disposers.forEach(dispose => dispose());
+    return () => {
+      abortController.abort();
+    };
+  }
+
+  private watchNamespace(api: KubeApi<T>, namespace: string, abortController: AbortController) {
+
+    const callback = (data: IKubeWatchEvent, error: any) => {
+      if (!this.isLoaded || abortController.signal.aborted) return;
+
+      if (error instanceof Response) {
+        if (error.status === 404) {
+          // api has gone, let's not retry
+          return;
+        } else { // not sure what to do, best to retry
+          setTimeout(() => {
+            if (abortController.signal.aborted) return;
```

Comment:
Already used in line 293, so doesn't make sense.

```ts
if (!this.isLoaded || abortController.signal.aborted) return;
```

Reply:
This happens somewhere in future and that's why it needs to be checked again, right?

Reply:
Why not just clear the timeout?

Reply:
No, because you get access to the same `abortController` from the scope above and if it was aborted the code below will be not reached?

Reply:
What I mean is why not (give line 300 having `const retry = setTimeout(...`) have the following:

```js
abortController.signal.addEventHandler("abort", () => clearTimeout(retry));
```

Reply:
@Nokel81 I like your suggestion, will cleanup.